src/HMRServer.js

```diff
@@ -1,10 +1,18 @@
+const http = require('http');
+const https = require('https');
 const WebSocket = require('ws');
 const prettyError = require('./utils/prettyError');
+const generateCertificate = require('./utils/generateCertificate');
 
 class HMRServer {
-  async start(port) {
+  async start(options = {}) {
     await new Promise(resolve => {
-      this.wss = new WebSocket.Server({port}, resolve);
+      let server = options.https
+        ? https.createServer(generateCertificate(options))
```

Comment:
Doesn't this result in generating two certificates? One for the HMR server and one for the normal web server? Perhaps we should share them...

Reply:
`generateCertificate` cache certificates. It seems I only get one in my .cache folder.

Reply:
Yes, the certificate is being cached.